libs/asset-viewer/src/lib/state/asset-search/asset-search.effects.ts

```diff
@@ -27,208 +26,160 @@ import {
 @Injectable()
 export class AssetSearchEffects {
   private readonly store = inject(Store<AppState>);
-  private readonly actions$ = inject(Actions);
+  private readonly actions$ = inject(Actions).pipe(share());
   private readonly router = inject(Router);
   private readonly route = inject(ActivatedRoute);
   private readonly assetSearchService = inject(AssetSearchService);
   private readonly allStudyService = inject(AllStudyService);
+  private readonly authService = inject(AuthService);
 
-  constructor() {
-    merge(this.store.select(fromAppShared.selectRDReferenceData))
-      .pipe(
-        filter(RD.isFailure),
-        map((e) => e.error),
-        filter(isDecodeError)
-      )
-      .subscribe((e) => {
-        console.error('DecodeError', D.draw(e.cause));
-      });
-  }
-
-  public loadSearch$ = createEffect(() =>
+  public updateQueryFromParams$ = createEffect(() =>
     this.actions$.pipe(
-      ofType(actions.search, actions.resetSearch, actions.removePolygon),
-      withLatestFrom(this.store.select(selectAssetSearchQuery)),
-      map(([_, query]) => actions.loadSearch({ query }))
+      ofType(actions.initialize),
+      switchMap(() => this.route.queryParams.pipe(take(1))),
+      switchMap((params) => {
+        const query: AssetSearchQuery = {};
+        const assetId = readNumberParam(params, QUERY_PARAM_MAPPING.assetId);
+        query.text = readStringParam(params, QUERY_PARAM_MAPPING.text);
+        query.polygon = readPolygonParam(params, QUERY_PARAM_MAPPING.polygon);
+        query.authorId = readNumberParam(params, QUERY_PARAM_MAPPING.authorId);
+        const min = readDateParam(params, QUERY_PARAM_MAPPING.createDate.min);
+        const max = readDateParam(params, QUERY_PARAM_MAPPING.createDate.max);
+        query.createDate = min && max ? { min, max } : undefined;
+        query.manCatLabelItemCodes = readArrayParam(params, QUERY_PARAM_MAPPING.manCatLabelItemCodes);
+        query.assetKindItemCodes = readArrayParam(params, QUERY_PARAM_MAPPING.assetKindItemCodes);
+        query.usageCodes = readArrayParam(params, QUERY_PARAM_MAPPING.usageCodes);
+        query.geometryCodes = readArrayParam(params, QUERY_PARAM_MAPPING.geometryCodes);
+        query.languageItemCodes = readArrayParam(params, QUERY_PARAM_MAPPING.languageItemCodes);
+        query.workgroupIds = readArrayParam<number>(params, QUERY_PARAM_MAPPING.workgroupIds);
+        return [
+          actions.search({ query }),
+          assetId == null ? actions.clearSelectedAsset() : actions.selectAsset({ assetId }),
+        ];
+      })
     )
   );
 
-  public assetResults$ = createEffect(() =>
+  public updateParamsFromQuery$ = createEffect(
+    () =>
+      combineLatest([
+        this.store.select(selectAssetSearchQuery),
```

Comment:
Even though this is valid, I prefer effects to be triggered by actions instead of state changes, as is intended by ngRx

Reply:
I'd argue it's in our best interest to make an exception here, as we want to always keep the params up-to-date with the query. Without reacting to state changes, this becomes much more cumbersome to track and quite easy to break.

Reply:
True. But I would say the NgRx way to do this would be to trigger this effect through any of the actions that change one of the required states. I do see, however, that this is more difficult to read, so it is fine by me